src/theory/fp/fp_converter.cpp

```diff
@@ -12,25 +12,1029 @@
  ** \brief Conversion of floating-point operations to bit-vectors using symfpu.
  **/
 
+// Only needed for the leaf test
+#include "theory/theory.h"
+
 #include "theory/fp/fp_converter.h"
 
 #include <stack>
 
+#include "symfpu/core/packing.h"
+#include "symfpu/core/sign.h"
+#include "symfpu/core/add.h"
+#include "symfpu/core/multiply.h"
+#include "symfpu/core/fma.h"
+#include "symfpu/core/sqrt.h"
+#include "symfpu/core/divide.h"
+#include "symfpu/core/compare.h"
+#include "symfpu/core/classify.h"
+#include "symfpu/core/convert.h"
+#include "symfpu/core/remainder.h"
+
+
+namespace symfpu {
+  using namespace ::CVC4::theory::fp::symfpuSymbolic;
+
+#define CVC4SYMITEDFN(T) template <>					\
+  struct ite<symbolicProposition, T> { \
+    static const T iteOp (const symbolicProposition &_cond,	\
+			  const T &_l,					\
+			  const T &_r) {				\
+      ::CVC4::NodeManager* nm = ::CVC4::NodeManager::currentNM();	\
+      									\
+      ::CVC4::Node cond = _cond.getNode();				\
+      ::CVC4::Node l = _l.getNode();					\
+      ::CVC4::Node r = _r.getNode();					\
+									\
+      /* Handle some common symfpu idioms */				\
+      if (cond.isConst()) {						\
+	if (cond == nm->mkConst(::CVC4::BitVector(1U, 1U))) {		\
+	  return l;							\
+	} else {							\
+	  return r;							\
```

Comment:
Again, you could use the shorthand `... ? ... : ... ;` here.

Reply:
and so I have!